### ISSUE TITLE
ServerBuilder should accept a Server header configuration setting

### ISSUE THREAD

**Issue by bpholt** [http4s/http4s#1905]
Currently it's inconvenient to create a server that includes a `Server` header on every response. It should be possible to configure this on `ServerBuilder` instead of using middleware to add the header.

**Comment by rossabaker** [comment]
I don't think we encoded our server builders in such a way that we can add fields in a binary compatible way in 0.18.  Refactoring to something similar to what Contraband generates is something I'd like to do, and undo the maze of `ServerBuilder` related traits.  #1882, is a rough sketch of how I imagine clients and servers looking in the future.  Tedious to write, but easier to evolve.

That said, I'm for this change happening somehow, and if it's just by following the existing patterns, I'm okay with that, and we can make sure it's preserved in any subsequent refactoring.

(Everything in this ticket probably applies to client `User-Agent`s as well.)

**Comment by rossabaker** [comment]
The builders are refactored.  We just need to do the work.
